src/redux/actions/applicationActions.test.ts

```diff
@@ -61,4 +69,36 @@ describe("Application Redux Actions", () => {
 
         expect(result).toEqual(appSettings);
     });
+
+    it("Ensure security token action creates a token if one doesn't exist", async () => {
+        const appSettings: IAppSettings = {
+            devToolsEnabled: false,
+            securityTokens: [
+                { name: "A", key: "1" },
+                { name: "B", key: "2" },
+                { name: "C", key: "3" },
+            ],
+        };
+        const middleware = [thunk];
+        const mockState: IApplicationState = {
+            ...initialState,
+            appSettings,
+        };
+
+        store = createMockStore<IApplicationState>(middleware)(mockState);
+
+        const testProject = MockFactory.createTestProject("TestProject");
+
+        const result = await applicationActions.ensureSecurityToken(testProject)(store.dispatch, store.getState);
+        const actions = store.getActions();
+
+        expect(actions.length).toEqual(1);
+        expect(actions[0]).toEqual({
+            type: ActionTypes.ENSURE_SECURITY_TOKEN_SUCCESS,
+            payload: testProject,
+        });
+
+        expect(result).toEqual(appSettings);
+        expect(testProject.securityToken).toEqual("Project TestProject Token");
```

Comment:
I would also confirm that the security token was correctly added into the appSettings.securityTokens array.